src/components/KCheckbox/KCheckbox.vue

```diff
@@ -121,45 +143,139 @@ export default {
 </script>
 
 <style lang="scss" scoped>
+/* Component mixins */
+
+@mixin kCheckboxIcon {
+  inset: 0;
+  left: calc(50% - 2.4px); // 2px is not enough, 3px is too much...
+  pointer-events: none;
+  position: absolute;
+  top: calc(50% + 1.75px); // 1px is not enough, 2px is too much...
+  transform: translate(-50%, -50%);
+  z-index: 1;
+}
 
-@import '@/styles/mixins';
+/* Component styles */
 
 .k-checkbox {
-  .k-input {
-    @include input-type-checkbox;
+  align-items: center;
+  display: inline-flex;
+
+  &.has-description {
+    align-items: flex-start;
+
+    .checkbox-input-wrapper {
+      margin-top: 3px; // align with label
+    }
+  }
+
+  .checkbox-input-wrapper {
+    display: flex;
+    position: relative;
+  }
+
+  /* Checkbox styles */
+  .checkbox-input {
+    @include radioCheckboxDefaults;
+
+    border-radius: var(--kui-border-radius-20, $kui-border-radius-20);
+
+    &:hover {
+      @include radioCheckboxHover;
+    }
+
+    &:focus-visible {
+      @include radioCheckboxFocus;
+    }
+
+    &:active:not(:disabled) {
+      @include radioCheckboxActive;
+    }
+
+    &:checked, &:indeterminate {
+      @include radioCheckboxChecked;
+
+      &:focus-visible {
+        @include radioCheckboxCheckedFocus;
+      }
+
+      &:active {
+        @include radioCheckboxCheckedActive;
+      }
+
+      &:disabled {
+        @include radioCheckboxCheckedDisabled;
+      }
+    }
+
+    &:disabled {
+      @include radioCheckboxDisabled;
+    }
   }
 
-  .k-checkbox-label {
-    font-size: var(--kui-font-size-30, $kui-font-size-30);
-    font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular);
-    line-height: var(--kui-line-height-30, $kui-line-height-30);
-    margin: var(--kui-space-0, $kui-space-0);
-    vertical-align: middle;
+  &.input-error {
+    .checkbox-input {
+      &:not(:disabled) {
+        @include radioCheckboxError;
+
+        &:hover {
+          @include radioCheckboxErrorHover;
+        }
+
+        &:focus-visible {
+          @include radioCheckboxErrorFocus;
+        }
+
+        &:checked, &:indeterminate {
+          @include radioCheckboxErrorChecked;
+
+          &:focus-visible {
+            @include radioCheckboxErrorCheckedFocus;
+          }
+        }
+      }
+    }
   }
 
-  .k-checkbox-description {
-    color: var(--kui-color-text, $kui-color-text);
-    font-size: var(--kui-font-size-30, $kui-font-size-30);
-    font-weight: var(--kui-font-weight-regular, $kui-font-weight-regular);
-    line-height: var(--kui-line-height-30, $kui-line-height-30);
-    padding-left: var(--kui-space-80, $kui-space-80);
-    padding-top: var(--kui-space-20, $kui-space-20);
+  /* Check and indeterminate icon styles */
+  .checkbox-input:checked + .checkbox-icon,
+  .checkbox-input:indeterminate + .checkbox-icon {
+    @include kCheckboxIcon;
+
+    color: var(--kui-color-text-inverse, $kui-color-text-inverse) !important;
```

Comment:
same comment here about commenting if overriding the mixin

Reply:
Done